src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -60,11 +76,19 @@ public EditCommand parse(String args) throws ParseException {
         }
         parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
 
-        if (!editPersonDescriptor.isAnyFieldEdited()) {
-            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
+        if (argMultimap.getValue(PREFIX_MEDICAL).isPresent()) {
+            Collection<String> medicalHistoryValues = argMultimap.getAllValues(PREFIX_MEDICAL);
+            Set<MedicalHistory> medicalHistories = ParserUtil.parseMedicalHistories(medicalHistoryValues);
+            editPersonDescriptor.setMedicalHistories(medicalHistories);
+        }
+
+        if (argMultimap.getValue(PREFIX_START).isPresent() && argMultimap.getValue(PREFIX_END).isPresent()) {
+            Appointment appointment = ParserUtil.parseAppointment(
+                    argMultimap.getValue(PREFIX_START).get(), argMultimap.getValue(PREFIX_END).get());
+            editPersonDescriptor.setAppointment(appointment);
         }
```

Comment:
Can we separate this into its own method? The level of abstraction on this section does not match its surrounding code.

Reply:
It actually matches the surrounding code, just that the surrounding code combine all parts into a single line.
I can also combine it that way but it will make the code less readable.

Reply:
Ah, my mistake - I meant to cover only the `Set<MedicalHistory>` section. I point this out as the similar `Set<Tag>` uses an internal `parseTagsForEdit` method at `Line 71`.

Reply:
This is addressed in the latest commit.